### PR TITLE
add linter to go action

### LINKED ISSUE #14: Add tests to the dot
we need to add tests to existing CICD for DB connection
for discord ping
for twitter ping
simulate error responses

### PR DESCRIPTION
**Pull request by Soypete**
add test the repo and better Continuous integration steps

### CODE REVIEW

File: db/connection.go [hunk 1]
```diff
@@ -80,14 +78,23 @@ func (c *Connection) Migrate(ctx context.Context) {
 		log.Println("Table does not exist")
 		// insert data
 		result := c.DB.MustExecContext(ctx, videoInsert)
-		fmt.Println(result)
+		count, _ := result.RowsAffected()
```
Comment (MordFustang21) [inline db/connection.go]: Need to handle error

File: db/connection.go [hunk 2]
```diff
@@ -80,14 +78,23 @@ func (c *Connection) Migrate(ctx context.Context) {
...
+		if count > 0 {
+			log.Println("Inserted data")
+		}
 	} else {
-		row.Scan(&count)
+		err := row.Scan(&count)
```
Comment (MordFustang21) [inline db/connection.go:89]: Is there a rows.Close() somewhere?

File: main.go [hunk 3]
```diff
@@ -61,6 +73,10 @@ func main() {
 
 }
 
+// healthCheck is a http handler for health check to make sure the server is up.
 func healthCheck(w http.ResponseWriter, r *http.Request) {
-	w.Write([]byte("we are live"))
+	_, err := w.Write([]byte("we are live"))
+	if err != nil {
+		log.Fatal(err)
```
Comment (MordFustang21) [inline main.go]: Are you sure you want to kill the server just because you couldn't write a response to a client?

File: db/connection.go [hunk 4]
```diff
@@ -80,14 +78,23 @@ func (c *Connection) Migrate(ctx context.Context) {
...
+		if err != nil {
+			log.Println(err)
+		}
 		if count == 0 {
 			log.Println("Table is empty")
 			// insert data
 			result := c.DB.MustExecContext(ctx, videoInsert)
-			fmt.Println(result)
+			count, _ := result.RowsAffected()
```
Comment (MordFustang21) [inline db/connection.go]: Another error check

File: db/connection.go [hunk 5]
```diff
@@ -80,14 +78,23 @@ func (c *Connection) Migrate(ctx context.Context) {
...
+		if count > 0 {
+			log.Println("Inserted data")
+		}
 	} else {
-		row.Scan(&count)
+		err := row.Scan(&count)
+		if err != nil {
+			log.Println(err)
+		}
 		if count == 0 {
```
Comment (MordFustang21) [inline db/connection.go:93]: While not a part o this PR I think you could probably cleanup this if/else logic to eliminate the else and improve readability.

**Comment by MordFustang21** [comment]
You really should rebase/squash these commits

File: db/connection.go (hunk 1 above)
Comment (Soypete) [reply db/connection.go]: the linter didn't say I needed to... 😜

File: db/connection.go (hunk 2 above)
Comment (Soypete) [reply db/connection.go:89]: single row so no need to close

Comment (Soypete) [reply db/connection.go:89]: https://pkg.go.dev/database/sql#Row

File: main.go (hunk 3 above)
Comment (Soypete) [reply main.go]: Why? this is super unsuffisticated.

Comment (MordFustang21) [reply main.go]: If a client connection gets dropped or any other client error you don't want to take down the whole instance due to a bad client. The instance could still be in a good state with other requests in flight.

File: db/connection.go (hunk 5 above)
Comment (Soypete) [reply db/connection.go:93]: booo...  
Do be fair, this is super hacky and I want to refactor the entire migration sequence in the future

File: db/connection.go (hunk 2 above)
Comment (MordFustang21) [reply db/connection.go:89]: my bad didn't see it was a row not rows.

File: .github/workflows/go.yml [hunk 6]
```diff
@@ -8,18 +8,35 @@ on:
...
+    environment: CICD
+
     steps:
     - uses: actions/checkout@v3
 
     - name: Set up Go
       uses: actions/setup-go@v2
       with:
-        go-version: 1.17
+        go-version: 1.18
```
Comment (jnewmano) [inline .github/workflows/go.yml:21]: The go.mod file has `1.16`, should they match each other?

File: .github/workflows/golangci-lint.yml [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+  contents: read
+  pull-requests: read
+
+jobs:
+
+  golangci:
+    strategy:
+      matrix:
+        go: [1.18]
+        os: [macos-latest, windows-latest]
```
Comment (jnewmano) [inline .github/workflows/golangci-lint.yml]: drop windows?

File: main.go (hunk 3 above)
Comment (jnewmano) [reply main.go]: I would either ignore the error with an `_` or just log a message, I wouldn't exit. Given that this is a health check endpoint, it's up to whatever is running the app to decide what to do if it gets an error response back from the app.

File: db/connection.go (hunk 5 above)
Comment (Soypete) [reply db/connection.go:93]: the whole things needs to be gutted and redone

**Review by jnewmano — APPROVED** [review]
Good addition of linter, will lead to improved contributions moving forward.

**Pull request merged**